assets/js/googlesitekit/modules/datastore/sharing-settings.test.js

```diff
@@ -612,5 +616,143 @@ describe( 'core/modules sharing-settings', () => {
 				).toBe( false );
 			} );
 		} );
+
+		describe( 'haveSharingSettingsExpanded', () => {
+			it( 'requires the `key` param', () => {
+				expect( () => {
+					registry
+						.select( CORE_MODULES )
+						.haveSharingSettingsExpanded( 'invalid-param' );
+				} ).toThrow( 'key must be one of: management, sharedRoles.' );
+			} );
+
+			it( 'should return undefined if `sharingSettings` or `savedSharingSettings` cannot be loaded', () => {
+				global[ dashboardSharingDataBaseVar ] = undefined;
+
+				expect(
+					registry
+						.select( CORE_MODULES )
+						.haveSharingSettingsExpanded( 'management' )
+				).toBeUndefined();
+			} );
+
+			it( 'informs whether the `management` setting for any module has been changed from `owner` to `all_admins`', async () => {
+				global[ dashboardSharingDataBaseVar ] = undefined;
+
+				// Initially undefined.
+				expect(
+					registry
+						.select( CORE_MODULES )
+						.haveSharingSettingsExpanded( 'management' )
+				).toBeUndefined();
+
+				global[ dashboardSharingDataBaseVar ] = dashboardSharingData;
+				registry.select( CORE_MODULES ).getSharingSettings();
+
+				// Still false after getting the sharing settings from the global variable.
+				expect(
+					registry
+						.select( CORE_MODULES )
+						.haveSharingSettingsExpanded( 'management' )
+				).toBe( false );
+
+				// True after updating module's `management` from `owner` to `all_admins` on the client.
+				registry
+					.dispatch( CORE_MODULES )
+					.setSharingManagement( 'analytics', 'all_admins' );
+				expect(
+					registry
+						.select( CORE_MODULES )
+						.haveSharingSettingsExpanded( 'management' )
+				).toBe( true );
+
+				// False after updating module's `management` back to original server value on client.
+				registry
+					.dispatch( CORE_MODULES )
+					.setSharingManagement( 'analytics', 'owner' );
+				expect(
+					registry
+						.select( CORE_MODULES )
+						.haveSharingSettingsExpanded( 'management' )
+				).toBe( false );
+
+				// False after updating module's `management` from `all-admins` to `owner`.
+				registry
+					.dispatch( CORE_MODULES )
+					.setSharingManagement( 'search-console', 'owner' );
+				expect(
+					registry
+						.select( CORE_MODULES )
+						.haveSharingSettingsExpanded( 'management' )
+				).toBe( false );
+			} );
+
+			it( "informs whether the `sharedRoles` settings for any module contain roles that haven't been previously selected", async () => {
```

Comment:
Again, no need for `async` here

```suggestion
			it( "informs whether the `sharedRoles` settings for any module contain roles that haven't been previously selected", () => {
```